e2e/contracts/testdappv2/TestDAppV2.sol

```diff
@@ -225,9 +237,19 @@ contract TestDAppV2 {
     // Callable interface on connected EVM chains
     function onCall(MessageContext calldata messageContext, bytes calldata message) external payable returns (bytes memory) {
         string memory messageStr = message.length == 0 ? getNoMessageIndex(messageContext.sender) : string(message);
+        
+        IERC20 zetaTokenContract = IERC20(zetatoken);
+        uint256 allowance = zetaTokenContract.allowance(msg.sender, address(this));
+
+        uint256 amountToRecord = msg.value; 
+
+        if (allowance > 0) {
+            require(zetaTokenContract.transferFrom(msg.sender, address(this), allowance), "Zetatoken transfer from gateway failed");
+            amountToRecord = allowance; // Use token amount instead of ETH
```

Comment:
This is the change added to consume the tokens allocated to mainting the Mint vs Supply check

Reply:
It would be great if you could add comment and saying the `amountToRecord` is never used/checked by E2E the `eth_withdraw_and_call` tests, so it does not matter to overwrite the the `amountToRecord` value; and the purpose is to transfer away all allowed (from gateway) ZETA.